lib/orderbook/TradingPair.ts

```diff
@@ -300,13 +301,18 @@ class TradingPair {
     while (remainingOrder && !queue.isEmpty()) {
       // get the best available maker order from the top of the queue
       const makerOrder = queue.peek()!;
-      const matchingQuantity = getMatchingQuantity(remainingOrder, makerOrder);
+      const makerAvailableQuantityOrder = isOwnOrder(makerOrder)
+        ? { ...makerOrder, quantity: makerOrder.quantity - makerOrder.hold, hold: 0 }
+        : { ...makerOrder };
```

Comment:
Can we just set this equal to `makerOrder` instead of a copy of `makerOrder`? It seems like shouldn't be an issue, and it should make this slightly more performant.

Reply:
I started with it to be consistent with having an immutable object on both cases, but in the end we're only mutating `makerOrder` and not `makerAvailableQuantityOrder`, so it can indeed be removed.